lib/resty/etcd/v3.lua

```diff
@@ -92,16 +87,19 @@ local function http_request_uri(self, http_cli, method, uri, body, headers, keep
     })
 
     if err then
-        if health_check.conf ~= nil then
+        if health_check.conf.shm_name ~= nil then
             health_check.report_failure(endpoint.http_host)
-            err = endpoint.http_host .. ": " .. err
+        else
+            health_check.report_round_robin_target_failure(endpoint.http_host)
         end
-        return nil, err
+        return nil, endpoint.http_host .. ": " .. err
```

Comment:
can we move `if ... else ...` to health_check?

Reply:
updated